src/parser/deserializer.ts

```diff
@@ -284,9 +285,11 @@ function deserializeObject(
         (buffer[index++] << 16) |
         (buffer[index++] << 24);
     } else if (elementType === constants.BSON_DATA_NUMBER && promoteValues === false) {
+      const dataview = new DataView(buffer.buffer, buffer.byteOffset, buffer.byteLength);
```

Comment:
The thinking with declaring the Dataview at the top of the loop is that numbers are a very common data type so paying the allocation cost once and sharing it for the rest of the document's processing out weighs nesting it here and potentially allocating and garbage collecting it for every number processed. Are you able to gain any insight into that concern? Either through benching or possibly the chrome dev tools may show information related to GC

Reply:
When I have time I will create a Benchmark based on our most used mongodb document and see how much improvement it makes on it